core/testing.go

```diff
@@ -1,40 +1,59 @@
 package core
 
 import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
 	"github.com/tendermint/tendermint/abci/example/kvstore"
 	"github.com/tendermint/tendermint/abci/types"
 	"github.com/tendermint/tendermint/node"
 	rpctest "github.com/tendermint/tendermint/rpc/test"
 )
 
-const defaultRetainBlocks int64 = 10
+const defaultRetainBlocks int64 = 50
+
+// StartTestNode starts a mock Core node background process and returns it.
+func StartTestNode(t *testing.T, app types.Application) *node.Node {
+	nd := rpctest.StartTendermint(app, rpctest.SuppressStdout, rpctest.RecreateConfig)
+	t.Cleanup(func() {
+		rpctest.StopTendermint(nd)
+	})
+	return nd
+}
 
-// StartMockNode starts a mock Core node background process and returns it.
-func StartMockNode(app types.Application) *node.Node {
-	return rpctest.StartTendermint(app, rpctest.SuppressStdout, rpctest.RecreateConfig)
+// StartTestKVApp starts Tendermint KVApp.
+func StartTestKVApp(t *testing.T) (*node.Node, types.Application) {
+	app := CreateKVStore(defaultRetainBlocks)
+	return StartTestNode(t, app), app
 }
 
-// CreateKvStore creates a simple kv store app and gives the user
+// CreateKVStore creates a simple kv store app and gives the user
 // ability to set desired amount of blocks to be retained.
-func CreateKvStore(retainBlocks int64) *kvstore.Application {
+func CreateKVStore(retainBlocks int64) *kvstore.Application {
 	app := kvstore.NewApplication()
 	app.RetainBlocks = retainBlocks
 	return app
 }
 
-// StartRemoteClient returns a started remote Core node process, as well its
+// StartTestClient returns a started remote Core node process, as well its
 // mock Core Client.
-func StartRemoteClient() (*node.Node, Client, error) {
-	remote := StartMockNode(CreateKvStore(defaultRetainBlocks))
-	protocol, ip := GetRemoteEndpoint(remote)
+func StartTestClient(t *testing.T) (*node.Node, Client) {
+	nd, _ := StartTestKVApp(t)
+	protocol, ip := GetEndpoint(nd)
 	client, err := NewRemote(protocol, ip)
-	return remote, client, err
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := client.Stop()
+		require.NoError(t, err)
+	})
+	err = client.Start()
+	require.NoError(t, err)
```

Comment:
So the client is now started and stopped automatically. Previously the func name was mentioning Start of the Client but wasn't really doing so. Also this simplifies tests as they dont need to copy boileplate lifecycling code